cypress/e2e/functional/document_tests/tiles_copy_test_spec.js

```diff
@@ -302,3 +307,46 @@ context('Test copy tiles from one document to other document', function () {
 
   });
 });
+
+context("Test copy tile within a document", function () {
+  it("Copies a graph tile within a document", function () {
+    beforeTest(student5);
+
+    // Add table tile and populate it with categorical data.
+    cy.log("Add table tile with categorical data");
+    clueCanvas.addTile("table");
+    cy.get(".primary-workspace").within((workspace) => {
+      tableToolTile.typeInTableCellXY(0, 0, "small");
+      tableToolTile.getTableCellXY(0, 0).should("contain", "small");
+      tableToolTile.typeInTableCellXY(1, 0, "medium");
+      tableToolTile.getTableCellXY(1, 0).should("contain", "medium");
+      tableToolTile.typeInTableCellXY(0, 1, "red");
+      tableToolTile.getTableCellXY(0, 1).should("contain", "red");
+      tableToolTile.typeInTableCellXY(1, 1, "green");
+      tableToolTile.getTableCellXY(1, 1).should("contain", "green");
+    });
+
+    // Graph the table data in a new graph tile
+    cy.get(".primary-workspace .tile-toolbar button.toolbar-button").eq(2).click();
```

Comment:
This method of getting the "Graph It" button isn't ideal but seems to be the only way to do it without changes to the toolbar buttons.

I tried `clueCanvas.clickToolbarButton("table", "data-set-view")` instead, but that resulted in a click on the table tile's "DataCard It" button since it also has the `data-set-view` class.

Should the "Graph It" button have a different class? Or perhaps an additional class,  like `data-set-view-graph`? (Currently the "Graph It" and "DataCard It" buttons both have only the classes `toolbar-button` and `data-set-view`.)

Or should we maybe add `data-testid` attributes to the buttons and use those instead of classes to access the elements?

Reply:
You could also use the `data-original-title` attribute of the div that wraps the button. That would give you the "Graph It!" text that you could match. There is a helper `getGraphItButton` in DataCardToolTile.js which does this.

But yes if you want to add data-testid to all of the buttons that would be good too. For these data-set-view buttons they'll have to be dynamic based on what tile type the data-set-view button is configured to create.

Reply:
Thanks, @scytacki . I did try that yesterday but it didn't work. I tried it again to confirm I wasn't doing something wrong somehow before. I found that even though the `data-original-title` value is "graph It!" in my browser, when the test runs in Cypress, the value is "Graph It!". That's one weird thing. It gets more interesting in that after I updated the test to use `cy.get(".modal-button").contains("Graph It").click()` and that worked, I thought maybe it'd be better to add a `data-test-id` anyway, but when I did, all of a sudden in Cypress the button had a class of `link-graph` instead of `data-set-view`. It still has the latter value in my browser, but in Cypress it continually has `link-graph` now. I'm pretty sure I made no changes that could have caused that.

There is a `link-graph` button in [src/components/tiles/table/table-toolbar-registration.tsx](https://github.com/concord-consortium/collaborative-learning/blob/2484f7f7dd9098e9f626f70c2e5f0a4af277654b/src/components/tiles/table/table-toolbar-registration.tsx#L126-L129) which should use `LinkGraphButton` instead of `DataSetViewButton`, but it's not clear to me why that might get used in some cases but not others. Do you have any sense of why that might be?

I did update the test to use `cy.get(".primary-workspace .tile-toolbar button.toolbar-button.link-graph").click();` for now because I'm curious to see if it'll pass here like it's passing for me locally. I'm thinking that's probably not going to be the final fix though.

Reply:
On reviewing this again, it seems like the differences I'm seeing are due to the test using a different `unit` URL param than what I'm using to test manually in the browser. So it seems like using `cy.get("[data-original-title='Graph It!']").click()` should be sufficient to finish this PR.

Also, there's a typo in my previous comment. Sorry, I should've written `cy.get("[data-original-title='Graph It!']").click()` where I wrote `cy.get(".modal-button").contains("Graph It").click()`.